addons/radar/functions/fnc_cacheLoop.sqf

```diff
@@ -277,7 +279,7 @@ private _ctrlPosList = [0, 0, _listWidth * 10, _itemHeight * pixelH];
         _selected = format ["%1%2", (["", ">> "] select (_selectedUnits findIf {_x == _unit} > -1)), _num];
     };
     private _buddy = ["", _iconNamespace getVariable ["buddy", DUI_BUDDY]] select (_player == (_unit getVariable [QEGVAR(buddy,buddy), objNull]));
-    private _icon = [_unit getVariable [QGVAR(icon), DUI_RIFLEMAN], ""] select (_buddy != "" && {_onlyBuddyIcon});
+    private _icon = [[_unit getVariable [QGVAR(icon), DUI_RIFLEMAN], _unit getVariable QGVAR(speakingIcon)] select (_showSpeaking && {_unit getVariable [QGVAR(isSpeaking), false]}), ""] select (_buddy != "" && {_onlyBuddyIcon});
```

Comment:
This line is crazy. Is it an optimization thing? Can we … idk … add a few line breaks to clarify it a bit? like

```sqf
private _icon = [
    [
        _unit getVariable [QGVAR(icon), DUI_RIFLEMAN], 
        _unit getVariable QGVAR(speakingIcon)
    ] select (_showSpeaking && {_unit getVariable [QGVAR(isSpeaking), false]}), 
    ""
] select (_buddy != "" && {_onlyBuddyIcon});
```

Reply:
Good idea, yeah it is to avoid as many IF statements as possible, so it becomes wild with selects